spinalcordtoolbox/scripts/sct_label_vertebrae.py

```diff
@@ -66,6 +66,39 @@ def vertebral_detection_param(string):
     return param
 
 
+def parse_initz(string):
+    """Custom parser for -initz."""
+    try:
+        z, label = map(int, string.split(','))
+    except ValueError:
+        raise argparse.ArgumentTypeError(f'expected two integer values separated by a comma, got "{string}" instead')
+    return [z, label]
+
+
+class InitFileAction(argparse.Action):
+    """Custom action to handle the -initfile option."""
+    def __call__(self, parser, namespace, initfile, option_string=None):
+        # if user provided text file, parse and overwrite arguments
+        try:
+            args = open(initfile).read().split()
+        except OSError:
+            raise argparse.ArgumentError(self, f'error reading file "{initfile}"')
```

Comment:
> Is the goal with catching `OSErrors` here just to make sure we raise the error with the `--help` message? Or is it to provide a more helpful error message?

Both: a usage message together with the argument name `-initfile` and the problematic filename is friendlier and more useful than a pure Python traceback.

> Depending on the reasoning, it may be a little more foolproof to be agnostic about the error message/type:
> 
> ```python
>         try:
>             args = open(initfile).read().split()
>         except Exception as e:
>             raise argparse.ArgumentError(self, e)
> ```

For some reason I thought `except Exception` was frowned upon, but it's the recommended alternative to the (definitely frowned upon) bare `except`. TIL

I do like to be somewhat specific with my error types, though, and [`OSError`](https://docs.python.org/3/library/exceptions.html#OSError) is exactly the kind of error I intend to catch: a problem either opening or reading the file. If something else happens (like say, using an undefined variable) that would be a bug in the code rather than something the user can fix, and for those I think a pure Python traceback is appropriate.

> Also, just out of curiosity, I'm wondering if there's a meaningful distinction between `argparse.ArgumentError(self, msg)` and `parser.error(msg)`

There's a very slight difference! With `ArgumentError`, the error message is prefixed with the name of the command-line option which caused the problem. With `parser.error` you don't get this prefix. Personally I would default to using `ArgumentTypeError` inside a custom type function, `ArgumentError` inside a custom action, and `parser.error` for after-the-fact validation in the main function, if any.